### ISSUE TITLE
Wrong output shape with ValidRandomCrop

### ISSUE THREAD

**Issue by jbueltemeier** [pmeier/pystiche#169]
`ValidRandomCrop` sometimes does not seem to return the correct shape for odd image dimensions, like here 305. The shape is only one example where this error occured, but I was able to reproduce it for other odd entries as well. I put the minimal example into a loop, because the error does not occur in every pass, so it seems to be a random process. 

```python
import torch
from pystiche.image.transforms import ValidRandomCrop

image = torch.rand(1, 3, 350, 305)
transform = ValidRandomCrop((256, 256))
for i in range(10):
    output = transform(image)
    assert output.shape == (1, 3, 256, 256), (
        "Current shape is " + str(output.shape) + " instead of " + str((1, 3, 256, 256))
```

**Comment by pmeier** [comment]
Internally `ValidRandomCrop` uses `torch.randint()` to produce a pseudo random number:

https://github.com/pmeier/pystiche/blob/5e65e1fa50bdaa694054697c1228e8f1c4103453/pystiche/image/transforms/crop.py#L106

To make this reproducible you can set the random seed with [`torch.manual_seed()`](https://pytorch.org/docs/stable/torch.html#torch.manual_seed). Can you produce an example set random seed and without loop?

**Comment by jbueltemeier** [comment]
```python
import torch
from pystiche.image.transforms import ValidRandomCrop

torch.manual_seed(0)
image = torch.rand(1, 3, 350, 255)
transform = ValidRandomCrop((256, 256))

image = transform(image)
print("Current shape is " + str(image.shape) + " instead of " + str((1, 3, 256, 256)))
```

**Comment by pmeier** [comment]
You are trying to crop a `256 x 256` patch out of a `255 x 350` image. This is not going to work since a valid crop requires the image to be at least as large as the crop size. This was not a problem in your first example. Please find a valid example.

---

If this snippet runs for you, you are likely operating on an outdated master. #158 fixed a bug in `ValidCropTransform` that might already solve your problem.

**Comment by jbueltemeier** [comment]
The fixed bug in [#158](https://github.com/pmeier/pystiche/pull/158) already solve my problem. Thanks for your help.
